frontend/src/metabase/admin/databases/components/widgets/EngineWidget/EngineWidget.tsx

```diff
@@ -0,0 +1,336 @@
+          event.preventDefault();
+          break;
+      }
+    },
+    [field, activeIndex, activeOption, optionCount],
+  );
+
+  return (
+    <EngineSearchRoot role="combobox" aria-expanded="true">
+      <TextInput
+        value={searchText}
+        placeholder={t`Search for a database…`}
+        autoFocus
+        aria-autocomplete="list"
+        aria-controls={getListBoxId(rootId)}
+        aria-activedescendant={getListOptionId(rootId, activeOption)}
+        onChange={handleSearch}
+        onKeyDown={handleKeyDown}
+      />
+      {visibleOptions.length ? (
+        <EngineList
+          rootId={rootId}
+          options={visibleOptions}
+          activeIndex={activeIndex}
+          onOptionChange={field.onChange}
+        />
+      ) : (
+        <EngineEmptyState isHosted={isHosted} />
+      )}
+      {!isSearching && hasMoreOptions && (
+        <EngineToggle
+          isExpanded={isExpanded}
+          onExpandedChange={setIsExpanded}
+        />
+      )}
+    </EngineSearchRoot>
+  );
+};
+
+interface EngineListProps {
+  rootId: string;
+  options: EngineOption[];
+  activeIndex?: number;
+  onOptionChange?: (value: string) => void;
+}
+
+const EngineList = ({
+  rootId,
+  options,
+  activeIndex,
+  onOptionChange,
+}: EngineListProps): JSX.Element => {
+  return (
+    <EngineListRoot role="listbox" id={getListBoxId(rootId)}>
+      {options.map((option, optionIndex) => (
+        <EngineCard
+          key={option.value}
+          rootId={rootId}
+          option={option}
+          isActive={optionIndex === activeIndex}
+          onOptionChange={onOptionChange}
+        />
+      ))}
+    </EngineListRoot>
+  );
+};
+
+export interface EngineCardProps {
+  rootId: string;
+  option: EngineOption;
+  isActive: boolean;
+  onOptionChange?: (value: string) => void;
+}
+
+const EngineCard = ({
+  rootId,
+  option,
+  isActive,
+  onOptionChange,
+}: EngineCardProps): JSX.Element => {
+  const logo = getEngineLogo(option.value);
+
+  const handleClick = useCallback(() => {
+    onOptionChange?.(option.value);
+  }, [option, onOptionChange]);
+
+  return (
+    <EngineCardRoot
+      role="option"
+      id={getListOptionId(rootId, option)}
+      isActive={isActive}
+      onClick={handleClick}
+    >
+      {logo ? (
+        <EngineCardImage src={logo} />
+      ) : (
+        <EngineCardIcon name="database" />
+      )}
+      <EngineCardTitle>{option.name}</EngineCardTitle>
+    </EngineCardRoot>
+  );
+};
+
+interface EngineEmptyStateProps {
+  isHosted?: boolean;
+}
+
+const EngineEmptyState = ({ isHosted }: EngineEmptyStateProps): JSX.Element => {
+  return (
+    <EngineEmptyStateRoot>
+      <EngineEmptyIcon name="search" size={32} />
+      {isHosted ? (
+        <EngineEmptyText>{t`Didn’t find anything`}</EngineEmptyText>
+      ) : (
+        <EngineEmptyText>{jt`Don’t see your database? Check out our ${(
+          <ExternalLink href={Settings.docsUrl("developers-guide-drivers")}>
```

Comment:
I think you'll need to add a `key` prop to this `ExternalLink` since it is used within `jt`